src/core/meshcop/dataset_manager.cpp

```diff
@@ -321,6 +321,10 @@ void DatasetManager::HandleMgmtSetResponse(Coap::Message *aMessage, const Ip6::M
 exit:
     LogInfo("MGMT_SET finished: %s",
             error == kErrorNone ? StateTlv::StateToString(static_cast<StateTlv::State>(state)) : ErrorToString(error));
+    if (error == kErrorNone && state == StateTlv::kReject)
```

Comment:
One suggestion:

We can consider setting the `error ` after we successfully find and read the status TLV, i.e., after the ` VerifyOrExit(Tlv::Find<StateTlv>(*aMessage, state) == kErr....` line and not from `exit` section?
Something like:
```
   VerifyOrExit(Tlv::Find<StateTlv>(*aMessage, state) == kErrorNone && state != StateTlv::kPending,
                 error = kErrorParse);

    if (state == StateTlv::kReject)
    {
         error = kErrorRejected;
    }
```
This make the intention clear and may avoid code-checker warnings for `state` being used uninitialized?

Reply:
Sure, I was trying to preserve the logging message, it was "Reject" but now is "Rejected", not a problem for me.

I updated as suggested. And initialized state as `StateTlv::kPending` because if `Tlv::Find<StateTlv>(*aMessage, state)` fails then state may not be initialized.